error.go

```diff
@@ -33,6 +36,98 @@ func (e Error) GetID() string {
 	return e.ID
 }
 
+type ErrorLinks struct {
+	About string `json:"about,omitempty"`
+}
+
+type ErrorSource struct {
+	Pointer   string `json:"pointer,omitempty"`
+	Parameter string `json:"parameter,omitempty"`
+}
+
+func (e Error) MarshalJSON() ([]byte, error) {
+	// Collect the fields we need to marshal, all are optional.
+	var fields []string
+
+	if e.ID != "" {
+		fields = append(fields, fmt.Sprintf(`"id":%q`, e.ID))
+	}
+
+	if e.Links.About != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"links":{"about":%q}`, e.Links.About))
+	}
+
+	if e.Status != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"status":%q`, e.Status))
+	}
+
+	if e.Code != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"code":%q`, e.Code))
+	}
+
+	if e.Title != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"title":%q`, e.Title))
+	}
+
+	if e.Detail != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"detail":%q`, e.Detail))
+	}
+
+	if e.Source.Pointer != "" && e.Source.Parameter != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"source":{"pointer":%q,"parameter":%q}`,
+				e.Source.Pointer, e.Source.Parameter))
+	} else if e.Source.Pointer != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"source":{"pointer":%q}`, e.Source.Pointer))
+	} else if e.Source.Parameter != "" {
+		fields = append(fields,
+			fmt.Sprintf(`"source":{"parameter":%q}`, e.Source.Parameter))
+	}
+
+	// Capture our metadata using json.Marshal on the interface{}
+	if e.Meta != nil {
+		b, err := json.Marshal(e.Meta)
+		if err != nil {
+			return nil, err
+		}
+
+		fields = append(fields, `"meta":`+string(b))
+	}
+
+	// Generate our JSON
+	var buf bytes.Buffer
+	if _, err := buf.WriteString("{"); err != nil {
+		return nil, err
+	}
+
+	first := true
+	for _, field := range fields {
```

Comment:
I would recommend using the normal json.Marshal method instead of manually building a string:

```
package main

import (
  "encoding/json"
  "fmt"
  "log"
)

//Error struct tbd
type Error struct {
  ID     string      `json:"id,omitempty"`
  Links  ErrorLinks  `json:"links,omitempty"`
  Status string      `json:"status,omitempty"`
  Code   string      `json:"code,omitempty"`
  Title  string      `json:"title,omitempty"`
  Detail string      `json:"detail,omitempty"`
  Source ErrorSource `json:"source,omitempty"`
  Meta   interface{} `json:"meta,omitempty"`
}

//ErrorLinks struct tbd
type ErrorLinks struct {
  About string `json:"about,omitempty"`
}

//MarshalJSON overwrites the default bla
func (e ErrorLinks) MarshalJSON() ([]byte, error) {
  if e.About == "" {
    return []byte(`""`), nil
  }

  return json.Marshal(e.About)
}

// ErrorSource is used bla
type ErrorSource struct {
  Pointer   string `json:"pointer,omitempty"`
  Parameter string `json:"parameter,omitempty"`
}

func main() {
  links := ErrorLinks{}
  source := ErrorSource{}
  e := Error{ID: "id", Links: links, Status: "fisch", Source: source}
  result, err := json.Marshal(e)
  if err != nil {
    log.Fatal(err)
  }

  fmt.Printf("%#v", string(result))

}

```

Additionally please describe public structs to the linting "standard"  e.g.:

//Error struct can be used to......